<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Urinary Essential Elements of Young Children with Autism Spectrum Disorder (ASD) and their Mothers

- [Mendeley Reference](https://data.mendeley.com/datasets/79fk29dvk6/1)
- [Science Direct Reference](https://www.sciencedirect.com/science/article/abs/pii/S1750946720300088?via%3Dihub)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

### Table of Contents <a class="anchor" id="PAN_toc"></a>

* [Table of Contents](#PAN_toc)
    * [Page 1 - Abstract](#PAN_page_1)
    * [Page 2 - Imported Libraries](#PAN_page_2)
    * [Page 3 - Import the Dataset](#PAN_page_3)
    * [Page 4 - Setting Notebook Options](#PAN_page_4)
    * [Page 5 - Looking at the Data](#PAN_page_5)
    * [Page 6 - Checking the Column Names](#PAN_page_6)
    * [Page 7 - Cleaning the Column Names](#PAN_page_7)
    * [Page 8 - Creating a new Cleaned Dataset](#PAN_page_8)
    * [Page 9 - Counting Columns](#PAN_page_9)
    * [Page 10 - Get Info about the Dataset](#PAN_page_10)
    * [Page 11 - Get Descriptive Statistics about the Dataset](#PAN_page_11)
    * [Page 12 - Counting Rows and Removing any NANs](#PAN_page_12)
    * [Page 13 - Q1: "Is there a predictive relationship between levels of toxic and essential elements and the occurrence of ASD?"](#PAN_page_13)
    * [Page 14 - Q1 Analysis: Logistic Regressions](#PAN_page_14)
    * [Page 15 - Q1 Analysis: Logistic Regressions: Correlation Matrix](#PAN_page_15)
    * [Page 16 - Q1 Analysis: Logistic Regressions: Variance Inflaction Factor](#PAN_page_16)
    

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 1 - Abstract <a class="anchor" id="PAN_page_1"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

### Background
Even though the cause of autism spectrum disorders (ASD) remains unknown, the current understanding points towards complex interactions between environmental and genetic factors. One important environmental factor to consider is intake of toxic and essential elements, and their role in metabolism. Essential elements have received considerably less attention in the literature than the presence of toxins in urine.

### Method
The purpose of this investigation is to comprehensively assess the association between urinary element compositions of 28 mothers who had young children with ASD and 29 mothers who had young typically developing (TD) children, and in a subset of their children (21 with ASD and 26 TD).

### Questions
1. Is there a predictive relationship between levels of toxic and essential elements and the occurrence of ASD?
2. How does the urine test differ (Toxic vs Non-Toxic) for kids with Autism (ASD) to Typically Developing kids (TD)?
3. How does the urine test differ (Toxic vs Non-Toxic) for mothers with Autism (ASD) to Typically Developing moms (TD)?
4. How does the urine test differ (Toxic vs Non-Toxic) for kids with Autism (ASD) to their moms with Autism(ASD)?

### Analysis Setup
DV1 = Status with 2 levels (ASD or TD)<br>
DV2 = Type with 2 levels (Child or Mother)<br>
IV1 = Group of 20 Toxic elements (Aluminum to Uranium) <br>
IV2 = Group of 19 Essential elements (Boron to Creatinine)

### Results
...  

### Conclusions
...  


<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 2 - Imported Libraries<a class="anchor" id="PAN_page_2"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [121]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from statsmodels.stats.outliers_influence import variance_inflation_factor

#Will add additional as we go

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 3 - Import the Dataset <a class="anchor" id="PAN_page_3"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [122]:
data = pd.read_excel("../Data/DATA.xlsx")

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 4 - Setting Notebook Options<a class="anchor" id="PAN_page_4"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [123]:
# reset the options
pd.reset_option('display.max_rows')

# set the option to display the maximum number of columns
pd.set_option('display.max_columns', 50)

# set the option to display the maximum number of rows
pd.set_option('display.min_rows', 105)
pd.set_option('display.max_rows', 105)

pd.describe_option('display.max_rows')
pd.describe_option('display.max_columns')

display.max_rows : int
    If max_rows is exceeded, switch to truncate view. Depending on
    `large_repr`, objects are either centrally truncated or printed as
    a summary view. 'None' value means unlimited.

    In case python/IPython is running in a terminal and `large_repr`
    equals 'truncate' this can be set to 0 and pandas will auto-detect
    the height of the terminal and print a truncated object which fits
    the screen height. The IPython notebook, IPython qtconsole, or
    IDLE do not run in a terminal and hence it is not possible to do
    correct auto-detection.
    [default: 60] [currently: 105]
display.max_columns : int
    If max_cols is exceeded, switch to truncate view. Depending on
    `large_repr`, objects are either centrally truncated or printed as
    a summary view. 'None' value means unlimited.

    In case python/IPython is running in a terminal and `large_repr`
    equals 'truncate' this can be set to 0 or None and pandas will auto-detect
    the width o

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 5 - Looking at the Data<a class="anchor" id="PAN_page_5"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [124]:
data

'ID Number' 'Status'    'Type'  'Aluminum'  'Antimony'   'Arsenic'   
0              1    'ASD'   'Child'    7.000000    0.200000   11.000000  \
1              3    'ASD'   'Child'   19.000000    0.200000    8.100000   
2              5    'ASD'   'Child'    9.200000    0.000057    5.900000   
3              6    'ASD'   'Child'    8.400000    0.000057   14.000000   
4              8    'ASD'   'Child'    5.600000    0.000028   11.000000   
5             10    'ASD'   'Child'   17.000000    0.600000   15.000000   
6             12    'ASD'   'Child'    9.500000    0.000057  120.000000   
7             17    'ASD'   'Child'    6.800000    0.200000   35.000000   
8             19    'ASD'   'Child'   24.000000    0.200000   14.000000   
9             20    'ASD'   'Child'   21.000000    0.070000   12.000000   
10            25    'ASD'   'Child'    7.700000    0.100000   11.000000   
11            28    'ASD'   'Child'    3.200000    0.000057    0.002121   
12            29    'ASD'   'Child'    3.500000    0.000057   15.000000   
13            30    'ASD'   'Child'   18.000000    0.400000    4.200000   
14            31    'ASD'   'Child'   15.000000    0.000057    0.002121   
15            32    'ASD'   'Child'    3.500000    0.000057   12.000000   
16            33    'ASD'   'Child'   33.000000    0.000057    7.100000   
17            34    'ASD'   'Child'    5.600000    0.000057   12.000000   
18            35    'ASD'   'Child'    1.900000    0.000057   16.000000   
19            37    'ASD'   'Child'   15.000000    0.090000   11.000000   
20            39    'ASD'   'Child'   74.000000    0.300000   25.000000   
21            40     'TD'   'Child'   15.000000    0.000113   12.000000   
22            41     'TD'   'Child'   59.000000    0.300000   11.000000   
23            42     'TD'   'Child'   17.000000    0.000057   11.000000   
24            43     'TD'   'Child'   14.000000    0.000057    6.500000   
25            45     'TD'   'Child'    6.100000    0.000057   11.000000   
26            46     'TD'   'Child'    4.100000    0.000057   16.000000   
27            48     'TD'   'Child'   38.000000    0.000057   15.000000   
28            49     'TD'   'Child'   46.000000    0.900000    0.001061   
29            50     'TD'   'Child'    6.400000    0.000057    9.000000   
30            51     'TD'   'Child'   16.000000    0.300000   25.000000   
31            52     'TD'   'Child'   18.000000    0.200000   11.000000   
32            53     'TD'   'Child'   11.000000    0.200000   28.000000   
33            54     'TD'   'Child'   10.000000    0.000057   51.000000   
34            56     'TD'   'Child'   12.000000    0.080000    5.200000   
35            57     'TD'   'Child'   15.000000    0.000057   11.000000   
36            58     'TD'   'Child'   14.000000    0.000057   23.000000   
37            60     'TD'   'Child'    9.100000    0.200000   22.000000   
38            61     'TD'   'Child'   27.000000    0.200000   18.000000   
39            62     'TD'   'Child'    8.500000    0.100000   23.000000   
40            63     'TD'   'Child'    9.500000    0.200000   29.000000   
41            65     'TD'   'Child'    6.300000    0.090000    7.900000   
42            66     'TD'   'Child'   37.000000    0.200000   19.000000   
43            67     'TD'   'Child'    6.500000    0.100000   10.000000   
44            68     'TD'   'Child'    3.200000    0.100000   13.000000   
45            70     'TD'   'Child'   12.000000    0.000057   19.000000   
46            71     'TD'   'Child'   11.000000    0.200000   20.000000   
47             1    'ASD'  'Mother'    1.400000    0.000113   61.000000   
48             2    'ASD'  'Mother'    1.800000    0.000057    4.800000   
49             3    'ASD'  'Mother'    3.300000    0.000057    2.900000   
50             5    'ASD'  'Mother'    2.500000    0.000113   14.000000   
51             6    'ASD'  'Mother'    5.200000    0.000113    3.000000   
52             7    'ASD'  'Mo

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 6 - Checking the Column Names<a class="anchor" id="PAN_page_6"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [125]:
# check the column names - notice the extra ' single quotes
print(data.columns)

Index([''ID Number'', ''Status'', ''Type'', ''Aluminum'', ''Antimony'',
       ''Arsenic'', ''Barium'', ''Beryllium'', ''Bismuth'', ''Cadmium'',
       ''Cesium'', ''Gadolinium'', ''Lead'', ''Mercury'', ''Nickel'',
       ''Palladium'', ''Platinum'', ''Tellurium'', ''Thallium'', ''Thorium'',
       ''Tin'', ''Tungsten'', ''Uranium'', ''Boron'', ''Calcium'',
       ''Chromium'', ''Cobalt'', ''Copper'', ''Iron'', ''Lithium'',
       ''Magnesium'', ''Manganese'', ''Molybdenum'', ''Phosphorus'',
       ''Potassium'', ''Selenium'', ''Sodium'', ''Strontium'', ''Sulfur'',
       ''Vanadium'', ''Zinc'', ''Creatinine''],
      dtype='object')


<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 7 - Cleaning the Column Names<a class="anchor" id="PAN_page_7"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [126]:
# clean up the column names removing the extra '
data.columns = data.columns.str.replace("'", "")
# change the column name 'ID Number'
data.columns = data.columns.str.replace("ID Number", "ID")

In [127]:
# check the column names
print(data.columns)

Index(['ID', 'Status', 'Type', 'Aluminum', 'Antimony', 'Arsenic', 'Barium',
       'Beryllium', 'Bismuth', 'Cadmium', 'Cesium', 'Gadolinium', 'Lead',
       'Mercury', 'Nickel', 'Palladium', 'Platinum', 'Tellurium', 'Thallium',
       'Thorium', 'Tin', 'Tungsten', 'Uranium', 'Boron', 'Calcium', 'Chromium',
       'Cobalt', 'Copper', 'Iron', 'Lithium', 'Magnesium', 'Manganese',
       'Molybdenum', 'Phosphorus', 'Potassium', 'Selenium', 'Sodium',
       'Strontium', 'Sulfur', 'Vanadium', 'Zinc', 'Creatinine'],
      dtype='object')


<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 8 - Creating a new Cleaned Dataset<a class="anchor" id="PAN_page_8"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [128]:
data.to_csv("../Data/clean_data.csv", index=False)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 9 - Counting Columns<a class="anchor" id="PAN_page_8"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [129]:
# Count the number of columns
num_cols = data.shape[1]

print(num_cols)

42


<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 10 - Get Info about the Dataset<a class="anchor" id="PAN_page_8"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [130]:
# usually objects are your key factors/independent variables where floats and ints are continuous/dependent variables
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 105 entries, 0 to 104
Data columns (total 42 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   ID          105 non-null    int64  
 1   Status      105 non-null    object 
 2   Type        105 non-null    object 
 3   Aluminum    105 non-null    float64
 4   Antimony    105 non-null    float64
 5   Arsenic     105 non-null    float64
 6   Barium      105 non-null    float64
 7   Beryllium   105 non-null    float64
 8   Bismuth     105 non-null    float64
 9   Cadmium     105 non-null    float64
 10  Cesium      105 non-null    float64
 11  Gadolinium  105 non-null    float64
 12  Lead        105 non-null    float64
 13  Mercury     105 non-null    float64
 14  Nickel      105 non-null    float64
 15  Palladium   105 non-null    float64
 16  Platinum    105 non-null    float64
 17  Tellurium   105 non-null    float64
 18  Thallium    105 non-null    float64
 19  Thorium     105 non-null    f

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 11 - Get Descriptive Statistics about the Dataset<a class="anchor" id="PAN_page_11"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

## Descriptive statistics:
Basic descriptive statistics such as mean, median, mode, standard deviation, variance, range, and percentiles can be computed for each element to summarize its distribution.

In [131]:
# Calculate summary statistics for each element but drop "ID". Even though "ID" is numeric, we do not need stats on "ID"
stats = data.drop('ID', axis=1).describe()

# View the summary statistics
print(stats)

         Aluminum    Antimony     Arsenic      Barium   Beryllium     Bismuth   
count  105.000000  105.000000  105.000000  105.000000  105.000000  105.000000  \
mean     9.368598    0.059104   13.691641    2.718096    0.000458    0.315303   
std     12.035546    0.135363   16.684145    2.526504    0.000168    1.613558   
min      0.001414    0.000028    0.001061    0.000071    0.000177    0.000028   
25%      2.500000    0.000057    5.000000    1.100000    0.000354    0.000057   
50%      5.600000    0.000113   11.000000    1.900000    0.000354    0.000057   
75%     10.000000    0.070000   15.000000    3.300000    0.000707    0.000113   
max     74.000000    0.900000  120.000000   15.000000    0.000707   13.000000   

          Cadmium      Cesium  Gadolinium        Lead     Mercury      Nickel   
count  105.000000  105.000000  105.000000  105.000000  105.000000  105.000000  \
mean     0.084427    5.382857    0.315966    0.374105    0.104033    4.049527   
std      0.124575    3.3977

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">
           
# Page 12 - Counting Rows and Removing any NANs<a class="anchor" id="PAN_page_12"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

In [132]:
# count the number of rows using len()
num_rows = len(data)
print('Number of rows:', num_rows)

Number of rows: 105


In [133]:
# drop rows with missing values if there are any
data = data.dropna()

In [134]:
# count the number of rows using len()
num_rows = len(data)
print('Number of rows:', num_rows)

Number of rows: 105


<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 13 - Q1: "Is there a predictive relationship between levels of toxic and essential elements and the occurrence of ASD?" <a class="anchor" id="PAN_page_13"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

## Subgroup Analysis:
The data shows urine test result levels of toxic and non-toxic essential elements for children with ASD or TD type, as well as their ASD or TD type mothers. 
Logistic regression can help evaluate whether there is a significant association between the levels of elements and the occurrence of ASD, while controlling for potential confounding factors.

### Subgroups:

-child_subset  
-childToxic_subset (containing children ASD + TD only toxic elements)  
-childEssential_subset (containing children ASD + TD only non-toxic essential elements)  

-mother_subset  
-motherToxic_subset (containing Mother ASD + TD only toxic elements)  
-motherEssential_subset (containing Mother ASD + TD only non-toxic essential elements)  

## Creating child subsets:

In [135]:
## Create child subset:
child_subset = data.loc[data['Type'] == "'Child'"]
child_subset

ID Status     Type  Aluminum  Antimony     Arsenic  Barium  Beryllium   
0    1  'ASD'  'Child'       7.0  0.200000   11.000000     2.1   0.000707  \
1    3  'ASD'  'Child'      19.0  0.200000    8.100000     8.0   0.000354   
2    5  'ASD'  'Child'       9.2  0.000057    5.900000     4.7   0.000354   
3    6  'ASD'  'Child'       8.4  0.000057   14.000000     3.0   0.000354   
4    8  'ASD'  'Child'       5.6  0.000028   11.000000     3.8   0.000177   
5   10  'ASD'  'Child'      17.0  0.600000   15.000000     3.1   0.000354   
6   12  'ASD'  'Child'       9.5  0.000057  120.000000     4.2   0.000354   
7   17  'ASD'  'Child'       6.8  0.200000   35.000000     4.0   0.000707   
8   19  'ASD'  'Child'      24.0  0.200000   14.000000     2.6   0.000354   
9   20  'ASD'  'Child'      21.0  0.070000   12.000000     1.8   0.000354   
10  25  'ASD'  'Child'       7.7  0.100000   11.000000    10.0   0.000354   
11  28  'ASD'  'Child'       3.2  0.000057    0.002121     1.4   0.000354   
12  29  'ASD'  'Child'       3.5  0.000057   15.000000     4.2   0.000354   
13  30  'ASD'  'Child'      18.0  0.400000    4.200000     6.4   0.000354   
14  31  'ASD'  'Child'      15.0  0.000057    0.002121     0.5   0.000354   
15  32  'ASD'  'Child'       3.5  0.000057   12.000000     4.0   0.000354   
16  33  'ASD'  'Child'      33.0  0.000057    7.100000     2.8   0.000354   
17  34  'ASD'  'Child'       5.6  0.000057   12.000000     3.8   0.000354   
18  35  'ASD'  'Child'       1.9  0.000057   16.000000     5.4   0.000354   
19  37  'ASD'  'Child'      15.0  0.090000   11.000000     1.2   0.000354   
20  39  'ASD'  'Child'      74.0  0.300000   25.000000    15.0   0.000354   
21  40   'TD'  'Child'      15.0  0.000113   12.000000     1.6   0.000707   
22  41   'TD'  'Child'      59.0  0.300000   11.000000     7.2   0.000354   
23  42   'TD'  'Child'      17.0  0.000057   11.000000     5.2   0.000354   
24  43   'TD'  'Child'      14.0  0.000057    6.500000     7.4   0.000354   
25  45   'TD'  'Child'       6.1  0.000057   11.000000     2.9   0.000354   
26  46   'TD'  'Child'       4.1  0.000057   16.000000     2.8   0.000354   
27  48   'TD'  'Child'      38.0  0.000057   15.000000     3.3   0.000354   
28  49   'TD'  'Child'      46.0  0.900000    0.001061    11.0   0.000177   
29  50   'TD'  'Child'       6.4  0.000057    9.000000     2.5   0.000354   
30  51   'TD'  'Child'      16.0  0.300000   25.000000     7.0   0.000354   
31  52   'TD'  'Child'      18.0  0.200000   11.000000     1.0   0.000354   
32  53   'TD'  'Child'      11.0  0.200000   28.000000     9.8   0.000354   
33  54   'TD'  'Child'      10.0  0.000057   51.000000     6.8   0.000354   
34  56   'TD'  'Child'      12.0  0.080000    5.200000     3.0   0.000707   
35  57   'TD'  'Child'      15.0  0.000057   11.000000     1.7   0.000354   
36  58   'TD'  'Child'      14.0  0.000057   23.000000     2.7   0.000354   
37  60   'TD'  'Child'       9.1  0.200000   22.000000     1.7   0.000354   
38  61   'TD'  'Child'      27.0  0.200000   18.000000     1.9   0.000354   
39  62   'TD'  'Child'       8.5  0.100000   23.000000     2.9   0.000354   
40  63   'TD'  'Child'       9.5  0.200000   29.000000     5.5   0.000354   
41  65   'TD'  'Child'       6.3  0.090000    7.900000     1.8   0.000354   
42  66   'TD'  'Child'      37.0  0.200000   19.000000     2.0   0.000354   
43  67   'TD'  'Child'       6.5  0.100000   10.000000     4.0   0.000354   
44  68   'TD'  'Child'       3.2  0.100000   13.000000     2.8   0.000354   
45  70   'TD'  'Child'      12.0  0.000057   19.000000     5.2   0.000354   
46  71   'TD'  'Child'      11.0  0.200000   20.000000     2.1   0.000354   

     Bismuth   Cadmium  Cesium  Gadolinium  Lead   Mercury  Nickel  Palladium   
0   0.000113  0.000141     4.3    0.000099   0.2  0.000424     4.9   0.000849  \
1   0.000057  0.000071    16.0    0.000049   1.0  0.000212     2.9   0.000424   
2   0.200000  0.000071    11.0    0.000049   0.5  0.000212     9

In [136]:
# Recoding IV - Status into numeric
child_subset['Status'] = child_subset['Status'].replace({"'ASD'": 0, "'TD'": 1})

/var/folders/6v/tnqw_0q102z17p9kj6xrr6k80000gn/T/ipykernel_98637/3343021003.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  child_subset['Status'] = child_subset['Status'].replace({"'ASD'": 0, "'TD'": 1})


## Subsetting the Toxic Elements in Children:

In [137]:
# Create subset: child with Toxic elements only
childToxic_subset = child_subset.iloc[:, 0:23]
# Drop 'ID' and 'Type'
childToxic_subset = childToxic_subset.drop(columns=['ID', 'Type'])
childToxic_subset

Status  Aluminum  Antimony     Arsenic  Barium  Beryllium   Bismuth   
0        0       7.0  0.200000   11.000000     2.1   0.000707  0.000113  \
1        0      19.0  0.200000    8.100000     8.0   0.000354  0.000057   
2        0       9.2  0.000057    5.900000     4.7   0.000354  0.200000   
3        0       8.4  0.000057   14.000000     3.0   0.000354  0.000057   
4        0       5.6  0.000028   11.000000     3.8   0.000177  0.000028   
5        0      17.0  0.600000   15.000000     3.1   0.000354  0.000057   
6        0       9.5  0.000057  120.000000     4.2   0.000354  0.000057   
7        0       6.8  0.200000   35.000000     4.0   0.000707  0.000113   
8        0      24.0  0.200000   14.000000     2.6   0.000354  0.000057   
9        0      21.0  0.070000   12.000000     1.8   0.000354  0.000057   
10       0       7.7  0.100000   11.000000    10.0   0.000354  0.100000   
11       0       3.2  0.000057    0.002121     1.4   0.000354  0.000057   
12       0       3.5  0.000057   15.000000     4.2   0.000354  0.000057   
13       0      18.0  0.400000    4.200000     6.4   0.000354  0.000057   
14       0      15.0  0.000057    0.002121     0.5   0.000354  0.000057   
15       0       3.5  0.000057   12.000000     4.0   0.000354  0.000057   
16       0      33.0  0.000057    7.100000     2.8   0.000354  0.000057   
17       0       5.6  0.000057   12.000000     3.8   0.000354  0.000057   
18       0       1.9  0.000057   16.000000     5.4   0.000354  0.000057   
19       0      15.0  0.090000   11.000000     1.2   0.000354  0.000057   
20       0      74.0  0.300000   25.000000    15.0   0.000354  0.000057   
21       1      15.0  0.000113   12.000000     1.6   0.000707  0.000113   
22       1      59.0  0.300000   11.000000     7.2   0.000354  0.000057   
23       1      17.0  0.000057   11.000000     5.2   0.000354  0.000057   
24       1      14.0  0.000057    6.500000     7.4   0.000354  0.000057   
25       1       6.1  0.000057   11.000000     2.9   0.000354  0.000057   
26       1       4.1  0.000057   16.000000     2.8   0.000354  0.000057   
27       1      38.0  0.000057   15.000000     3.3   0.000354  0.000057   
28       1      46.0  0.900000    0.001061    11.0   0.000177  0.000028   
29       1       6.4  0.000057    9.000000     2.5   0.000354  0.000057   
30       1      16.0  0.300000   25.000000     7.0   0.000354  0.000057   
31       1      18.0  0.200000   11.000000     1.0   0.000354  0.000057   
32       1      11.0  0.200000   28.000000     9.8   0.000354  0.000057   
33       1      10.0  0.000057   51.000000     6.8   0.000354  0.000057   
34       1      12.0  0.080000    5.200000     3.0   0.000707  0.000113   
35       1      15.0  0.000057   11.000000     1.7   0.000354  0.600000   
36       1      14.0  0.000057   23.000000     2.7   0.000354  0.000057   
37       1       9.1  0.200000   22.000000     1.7   0.000354  0.000057   
38       1      27.0  0.200000   18.000000     1.9   0.000354  0.000057   
39       1       8.5  0.100000   23.000000     2.9   0.000354  0.000057   
40       1       9.5  0.200000   29.000000     5.5   0.000354  0.000057   
41       1       6.3  0.090000    7.900000     1.8   0.000354  0.200000   
42       1      37.0  0.200000   19.000000     2.0   0.000354  0.000057   
43       1       6.5  0.100000   10.000000     4.0   0.000354  0.000057   
44       1       3.2  0.100000   13.000000     2.8   0.000354  0.000057   
45       1      12.0  0.000057   19.000000     5.2   0.000354  0.400000   
46       1      11.0  0.200000   20.000000     2.1   0.000354  0.000057   

     Cadmium  Cesium  Gadolinium  Lead   Mercury  Nickel  Palladium  Platinum   
0   0.000141     4.3    0.000099   0.2  0.000424     4.9   0.000849  0.000113  \
1   0.000071    16.0    0.000049   1.0  0.000212     2.9   0.000424  0.000057   
2   0.000071    11.0    0.000049   0.5  0.000212     9.4   0.000424  0.000057   
3   0.000071    16.0    0.000049   0.9  0.000212    15.0   0.000424  0.000057  

## Subsetting the Essential Elements in Children:

In [138]:
# Create subset: child with Essential elements only
childEssential_subset = child_subset.iloc[:, 0:42]
# Drop 'ID' and 'Type'
childEssential_subset = childEssential_subset.drop(columns=['ID', 'Type'])
# Drop toxic elements
childEssential_subset = childEssential_subset.drop(columns=['Aluminum', 'Antimony', 'Arsenic', 'Barium',
       'Beryllium', 'Bismuth', 'Cadmium', 'Cesium', 'Gadolinium', 'Lead',
       'Mercury', 'Nickel', 'Palladium', 'Platinum', 'Tellurium', 'Thallium',
       'Thorium', 'Tin', 'Tungsten', 'Uranium'])
childEssential_subset

Status  Boron  Calcium  Chromium    Cobalt  Copper      Iron  Lithium   
0        0    2.3    140.0  0.000424  0.001131   0.014  0.240000    0.083  \
1        0    4.9     81.0  0.000212  0.000566   0.018  0.200000    0.170   
2        0    2.9    300.0  0.000600  0.003000   0.021  0.600000    0.170   
3        0    8.6    200.0  0.001000  0.000566   0.024  0.430000    0.061   
4        0    1.3     62.0  0.001000  0.000283   0.022  0.035355    0.220   
5        0    4.5    250.0  0.001000  0.000566   0.019  0.530000    0.120   
6        0    3.8    340.0  0.000500  0.001000   0.018  0.630000    0.083   
7        0    2.2    170.0  0.001000  0.000900   0.018  0.300000    0.140   
8        0    4.5    150.0  0.000212  0.000566   0.020  0.340000    0.084   
9        0    2.4     38.0  0.000400  0.000566   0.020  0.070711    0.055   
10       0    2.3    180.0  0.000212  0.000566   0.059  0.370000    0.044   
11       0    1.4     61.0  0.000212  0.000566   0.014  0.070711    0.100   
12       0    3.5    220.0  0.000212  0.000566   0.013  0.400000    0.066   
13       0    1.1    140.0  0.000300  0.000566   0.015  0.250000    0.054   
14       0    3.3     14.0  0.000212  0.000566   0.020  0.070711    0.043   
15       0    3.7     92.0  0.000212  0.000800   0.018  0.180000    0.084   
16       0    4.1     80.0  0.000212  0.001000   0.017  0.160000    0.064   
17       0    2.3    260.0  0.000212  0.002000   0.024  0.490000    0.350   
18       0    3.2    150.0  0.000212  0.000566   0.020  0.300000    0.110   
19       0    3.2     41.0  0.000212  0.000566   0.021  0.070711    0.058   
20       0   11.0    330.0  0.000700  0.001000   0.023  0.720000    0.110   
21       1    2.7     84.0  0.000424  0.001000   0.016  0.160000    0.150   
22       1    3.3    180.0  0.000212  0.000566   0.030  0.350000    0.130   
23       1    1.6    270.0  0.000212  0.000566   0.018  0.520000    0.230   
24       1    2.1    540.0  0.000212  0.001000   0.022  1.000000    0.099   
25       1    2.7    100.0  0.000212  0.000800   0.017  0.210000    0.092   
26       1    3.9     88.0  0.000212  0.003000   0.022  0.170000    0.099   
27       1    2.5    390.0  0.000212  0.002000   0.018  0.710000    0.170   
28       1    8.9    190.0  0.000106  0.000283   0.037  0.035355    0.130   
29       1    4.4     77.0  0.000212  0.000566   0.016  0.150000    0.071   
30       1    4.1    480.0  0.000212  0.000566   0.026  0.860000    0.300   
31       1    3.1     19.0  0.000400  0.000566   0.020  0.070711    0.180   
32       1    2.7    200.0  0.000212  0.000566   0.016  0.380000    0.073   
33       1    3.6    220.0  0.000212  0.000566   0.026  0.490000    0.160   
34       1    1.8     72.0  0.000424  0.001131   0.016  0.130000    0.037   
35       1    2.0    120.0  0.000212  0.000566   0.021  0.250000    0.120   
36       1    4.5    170.0  0.000212  0.002000   0.037  0.320000    0.140   
37       1    2.4     65.0  0.000212  0.001000   0.023  0.120000    0.140   
38       1    3.1     13.0  0.000212  0.000566   0.025  0.070711    0.120   
39       1    1.4     93.0  0.000212  0.000566   0.030  0.190000    0.190   
40       1    3.3    300.0  0.000212  0.002000   0.019  0.550000    0.240   
41       1    2.1    170.0  0.000212  0.000566   0.022  0.300000    0.066   
42       1    8.1     34.0  0.003000  0.000566   0.019  0.070711    0.067   
43       1    4.8    240.0  0.000500  0.001000   0.018  0.430000    0.180   
44       1    1.7    120.0  0.000212  0.000566   0.027  0.240000    0.075   
45       1    2.3    250.0  0.000212  0.001000   0.019  0.510000    0.060   
46       1    6.4     44.0  0.000600  0.000566   0.014  0.070711    0.150   

    Magnesium  Manganese  Molybdenum  Phosphorus  Potassium  Selenium  Sodium   
0         200     0.0007       0.130        1180         24     0.110      56  \
1         170     0.0050       0.180         360        120     0.094     290   
2         310     0.0040       0.170        1860        120     

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 14 - Q1 Analysis: Logistic Regressions <a class="anchor" id="PAN_page_13"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

## Logistic Regression of childToxic_subset Data:

In [139]:
# Split the data into predictors (X) and response variable (y)
X = childToxic_subset.drop('Status', axis=1)
y = childToxic_subset['Status']

# Create a logistic regression object
logreg = LogisticRegression(max_iter=1000)

# Fit the model using the training data
logreg.fit(X, y)

# Print the coefficients of the model
print(logreg.coef_)

[[ 8.59346988e-03  3.12459550e-01 -3.38920559e-03 -5.28382126e-02
  -1.03072593e-04  7.52768431e-02 -6.80419397e-02 -9.46182408e-02
  -2.25626804e-01 -8.36203582e-03 -6.92820820e-01  5.07390517e-02
  -1.23687112e-04 -1.64916149e-05 -8.24580745e-05 -1.33097005e-01
  -6.18435559e-06  3.78063436e-01  9.97516231e-01 -9.63085203e-02]]


The results show the coefficients for each predictor variable (toxic elements) in the logistic regression model. The coefficients represent the amount by which the log-odds of the response variable (ASD) changes with a one-unit increase in the corresponding predictor variable (toxic elements). For example, a positive coefficient for a predictor variable indicates that an increase in the value of that variable increases the log-odds of having ASD, while a negative coefficient indicates that an increase in the value of that variable decreases the log-odds of having an ASD diagnosis.

Further interpretation shows that an increase in the elements 'Aluminum', 'Antimony', 'Bismuth', 'Nickel', 'Tin', and 'Tungsten' increases the logistic odds of having ASD.

## Logistic Regression of childEssential_subset Data:

In [140]:
# Split the data into predictors (X) and response variable (y)
X = childEssential_subset.drop('Status', axis=1)
y = childEssential_subset['Status']

# Create a logistic regression object
logreg = LogisticRegression(max_iter=1000)

# Fit the model using the training data
logreg.fit(X, y)

# Print the coefficients of the model
print(logreg.coef_)

[[ 3.37310269e-01 -6.24911155e-03  2.93526147e-04  6.73010722e-06
   1.82762224e-04  6.86042301e-02 -6.19184710e-02 -1.48821223e-02
  -1.82552536e-03  1.68425309e-01  1.87883534e-03 -2.10592390e-02
  -7.01324644e-02  1.78896794e-03  1.55258021e-02  5.51426812e-03
   1.99495128e-04  7.81203603e-01  4.10116731e-03]]


The results provided are the coefficients for the logistic regression model using essential elements as predictors for TD status (typically developing). Similar to the previous interpretation, these coefficients represent the logistic odds of being in the TD group for each predictor(essential elements).

The first coefficient (0.337) represents the logistic odds of being in the TD group when the first predictor is increased by one unit, holding all other predictors constant. The second coefficient (-0.006) represents the log odds of being in the TD group when the second predictor is increased by one unit, holding all other predictors constant, and so on for each predictor.

Therefore, a positive coefficient suggests that an increase in the corresponding predictor variable is associated with higher odds of being in the TD group. Conversely, a negative coefficient suggests that an increase in the corresponding predictor variable is associated with lower odds of being in the TD group.

In this case, the coefficients suggest that higher levels of several essential elements are associated with higher odds of being in the TD group, including 'Boron', 'Iron', 'Molybdenum', and 'Zinc' predictors. Conversely, lower levels of several other essential elements are associated with higher odds of being in the TD group, including 'Magnesium', 'Manganese', and 'Selenium' predictors. However, it's important to keep in mind that interpreting the coefficients requires careful consideration of the specific context and other factors such as potential confounding variables.

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 15 - Q1 Analysis: Logistic Regressions: Correlation Matrix <a class="anchor" id="PAN_page_13"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

## Checking Multicollinearity with a Correlation Matrix 

The diagonal cells represent the correlation between an element and itself, which is always perfect and equal to 1.

The off-diagonal cells represent the correlation between two different elements, and can range from -1 to 1. A value of 1 represents a perfect positive correlation, where two elements increase or decrease together in a linear fashion. A value of -1 represents a perfect negative correlation, where two elements move in opposite directions. A value of 0 represents no correlation.

Interpretation of the correlation matrix depends on the context of the problem and the knowledge of the domain. However, in general, a correlation matrix can be used to identify which elements are positively or negatively correlated with each other, which can be helpful in understanding the underlying factors that affect the system being studied.

### childToxic_subset Data

In [141]:
# Select the columns that contain toxic element predictors
predictor_columns = ['Aluminum', 'Antimony', 'Arsenic', 'Barium',
       'Beryllium', 'Bismuth', 'Cadmium', 'Cesium', 'Gadolinium', 'Lead',
       'Mercury', 'Nickel', 'Palladium', 'Platinum', 'Tellurium', 'Thallium',
       'Thorium', 'Tin', 'Tungsten', 'Uranium']

# Calculate the correlation matrix
corr_matrix = childToxic_subset[predictor_columns].corr()

# Display the correlation matrix
print(corr_matrix)

            Aluminum  Antimony   Arsenic    Barium  Beryllium   Bismuth   
Aluminum    1.000000  0.478311 -0.064962  0.465886  -0.156897 -0.080045  \
Antimony    0.478311  1.000000 -0.104807  0.409503  -0.136278 -0.174034   
Arsenic    -0.064962 -0.104807  1.000000  0.103382   0.029799 -0.076867   
Barium      0.465886  0.409503  0.103382  1.000000  -0.224868 -0.060987   
Beryllium  -0.156897 -0.136278  0.029799 -0.224868   1.000000 -0.061363   
Bismuth    -0.080045 -0.174034 -0.076867 -0.060987  -0.061363  1.000000   
Cadmium     0.109310 -0.151251  0.376090 -0.073132  -0.074518  0.250531   
Cesium      0.295175  0.412028 -0.112565  0.418283  -0.390415 -0.092684   
Gadolinium -0.091334  0.066479  0.151660 -0.010550   0.455894 -0.042954   
Lead        0.699395  0.326466 -0.041431  0.620368  -0.234211 -0.112465   
Mercury    -0.149996 -0.155313 -0.129677  0.050639  -0.239100 -0.013001   
Nickel      0.047233 -0.040206  0.205503  0.323998   0.085808  0.132160   
Palladium  -0.156897 -0.1

These have "relatively high" correlations:  
Aluminum and Lead: 0.699395  
Antimony and Cesium: 0.412028  
Arsenic and Cadmium: 0.376090  
Barium and Lead: 0.620368  
Beryllium and Gadolinium: 0.455894  
Mercury and Lead: -0.623959  

### childEssential_subset Data

In [142]:
# Select the columns that contain essential element predictors
predictor_columns = ['Boron', 'Calcium', 'Chromium',
       'Cobalt', 'Copper', 'Iron', 'Lithium', 'Magnesium', 'Manganese',
       'Molybdenum', 'Phosphorus', 'Potassium', 'Selenium', 'Sodium',
       'Strontium', 'Sulfur', 'Vanadium', 'Zinc', 'Creatinine']

# Calculate the correlation matrix
corr_matrix = childEssential_subset[predictor_columns].corr()

# Display the correlation matrix
print(corr_matrix)

               Boron   Calcium  Chromium    Cobalt    Copper      Iron   
Boron       1.000000  0.081757  0.407813 -0.065262  0.098215  0.060715  \
Calcium     0.081757  1.000000 -0.120290  0.283998  0.099666  0.967354   
Chromium    0.407813 -0.120290  1.000000 -0.083071 -0.128955 -0.098354   
Cobalt     -0.065262  0.283998 -0.083071  1.000000 -0.023691  0.302262   
Copper      0.098215  0.099666 -0.128955 -0.023691  1.000000  0.063041   
Iron        0.060715  0.967354 -0.098354  0.302262  0.063041  1.000000   
Lithium    -0.076278  0.350803 -0.079157  0.261212  0.049876  0.310581   
Magnesium   0.250369  0.388201 -0.113281  0.302364  0.193352  0.312641   
Manganese   0.547624 -0.034174  0.145771 -0.171587  0.323571 -0.012884   
Molybdenum -0.168612  0.068217 -0.012842  0.276699 -0.073252  0.084909   
Phosphorus  0.000307  0.276708 -0.218089  0.164645  0.285396  0.132654   
Potassium   0.237380 -0.210503  0.168678 -0.187442  0.337587 -0.191672   
Selenium   -0.093593  0.233269 -0.0919

These have high correlations:  
Calcium and Magnesium have a correlation coefficient of 0.388201  
Calcium and Strontium have a correlation coefficient of 0.859805  
Iron and Calcium have a correlation coefficient of 0.967354  
Iron and Magnesium have a correlation coefficient of 0.312641  
Iron and Strontium have a correlation coefficient of 0.828568  
Manganese and Potassium have a correlation coefficient of 0.796127  
Boron and Manganese have a correlation coefficient of 0.547624  
Chromium and Vanadium have a correlation coefficient of 0.688543  
These correlations indicate that there may be some underlying relationships or dependencies between these variables. However, it is important to note that correlation does not necessarily imply causation, and further analysis would be needed to determine the nature of these relationships.

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

# Page 16 - Q1 Analysis: Logistic Regressions: Variance Inflaction Factor <a class="anchor" id="PAN_page_13"></a>

[Back to Top](#PAN_toc)

<hr style="height:5px;border-width:0;color:#D4AF37;background-color:#D4AF37">

## Checking Multicollinearity with Variance Inflation Factor (VIF)

Variance inflation factor (VIF) measures the extent to which the variance of an estimated regression coefficient is increased due to collinearity among the predictor variables. A VIF value of 1 indicates no correlation between a predictor variable and the other variables, while a value of 5 or above indicates a high degree of correlation.

### childToxic_subset Data

In [143]:
# Calculate VIF for each predictor
vif = pd.DataFrame()
vif["VIF Factor"] = [variance_inflation_factor(childToxic_subset.drop("Status", axis=1).values, i) for i in range(childToxic_subset.shape[1]-1)]
vif["Predictor"] = childToxic_subset.columns.drop("Status")

# display the VIFs
print(vif)

    VIF Factor   Predictor
0     6.765832    Aluminum
1     3.029613    Antimony
2     2.768722     Arsenic
3     6.837791      Barium
4          inf   Beryllium
5     1.566647     Bismuth
6     1.864104     Cadmium
7    16.204812      Cesium
8     1.439292  Gadolinium
9     8.631501        Lead
10    1.477657     Mercury
11    8.322813      Nickel
12         inf   Palladium
13         inf    Platinum
14         inf   Tellurium
15    8.971717    Thallium
16         inf     Thorium
17    3.275277         Tin
18    2.320321    Tungsten
19    1.390332     Uranium


/usr/local/Cellar/jupyterlab/3.6.3/libexec/lib/python3.11/site-packages/statsmodels/stats/outliers_influence.py:195: RuntimeWarning: divide by zero encountered in scalar divide
  vif = 1. / (1. - r_squared_i)
/usr/local/Cellar/jupyterlab/3.6.3/libexec/lib/python3.11/site-packages/statsmodels/stats/outliers_influence.py:195: RuntimeWarning: divide by zero encountered in scalar divide
  vif = 1. / (1. - r_squared_i)
/usr/local/Cellar/jupyterlab/3.6.3/libexec/lib/python3.11/site-packages/statsmodels/stats/outliers_influence.py:195: RuntimeWarning: divide by zero encountered in scalar divide
  vif = 1. / (1. - r_squared_i)
/usr/local/Cellar/jupyterlab/3.6.3/libexec/lib/python3.11/site-packages/statsmodels/stats/outliers_influence.py:195: RuntimeWarning: divide by zero encountered in scalar divide
  vif = 1. / (1. - r_squared_i)
/usr/local/Cellar/jupyterlab/3.6.3/libexec/lib/python3.11/site-packages/statsmodels/stats/outliers_influence.py:195: RuntimeWarning: divide by zero encountered in s

Based on the VIF values, it seems that Beryllium, Palladium, Platinum, and Tellurium have infinite VIF values, which could indicate that these variables are perfectly correlated with one or more of the other variables in the model. This should be investigated further.

Variables with high VIF values are Aluminum, Barium, Cesium, Lead, Nickel, and Thallium, which suggests that these variables are highly correlated with one or more of the other variables in the model. If this is the case, it could be problematic for the model since it may cause issues with model interpretation and stability. It's important to identify the source of the multicollinearity and consider ways to address it, such as removing one or more of the correlated variables or using a different modeling approach.

On the other hand, variables with low VIF values, such as Bismuth, Cadmium, Gadolinium, Mercury, Tin, Tungsten, and Uranium, have lower degrees of multicollinearity with other variables and may be good predictors in the model.

### childEssential_subset Data

In [144]:
# calculate VIF for each predictor
vif = pd.DataFrame()
vif["VIF Factor"] = [variance_inflation_factor(childEssential_subset.drop("Status", axis=1).values, i) for i in range(childEssential_subset.shape[1]-1)]
vif["Predictor"] = childEssential_subset.columns.drop("Status")


# display the VIFs
print(vif)

    VIF Factor   Predictor
0    14.188669       Boron
1    99.793089     Calcium
2    15.459840    Chromium
3     5.284753      Cobalt
4    17.190751      Copper
5    91.410496        Iron
6    10.607305     Lithium
7    20.894110   Magnesium
8    23.280519   Manganese
9     8.887762  Molybdenum
10   17.569043  Phosphorus
11   22.988763   Potassium
12   46.454833    Selenium
13   16.702465      Sodium
14   46.706092   Strontium
15   58.448534      Sulfur
16   28.509322    Vanadium
17   13.039539        Zinc
18    4.764125  Creatinine


Boron, Chromium, Cobalt, Lithium, Molybdenum, Phosphorus, Sodium, and Zinc all have VIF values below 10, indicating a low degree of correlation. However, the predictors Calcium, Copper, Iron, Magnesium, Manganese, Potassium, Selenium, Strontium, Sulfur, and Vanadium all have VIF values above 10, indicating a high degree of correlation.

It's important to note that while a high VIF value is often interpreted as evidence of collinearity, it does not necessarily imply causality. If you observe a high VIF value for a predictor, it could be due to the influence of other variables, rather than the predictor itself. In such cases, it's important to assess the nature of the correlation between the variables and consider potential solutions such as dropping the highly correlated predictors or performing principal component analysis to reduce the number of variables.